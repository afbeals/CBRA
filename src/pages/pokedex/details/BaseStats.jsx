// Extenal
import React from 'react';
import PropTypes from 'prop-types';

// Constants
const classname = 'base';

// Component
const BaseStats = ({ pClassname, height, weight, types }) => (
  <div
    data-testid="details-baseStats"
    className={`${pClassname}__${classname}`}
  >
    <div className={`${pClassname}__${classname}__info`}>
      <div className={`section section--info`}>
        <span className={'section__title'}>height:</span>
        <p className={'section__content'}>{height}</p>
      </div>
      <div className={`section section--info`}>
        <span className={'section__title'}>weight:</span>
        <p className={'section__content'}>{weight}</p>
      </div>
      <div className={`section section--types`}>
        {types.map(({ type: { name: type } }, i) => (
          <div key={`${type}-${i}`} className={`tag ${type}`}>
            {type}
          </div>
        ))}
      </div>
    </div>
  </div>
);

BaseStats.propTypes = {
  pClassname: PropTypes.string.isRequired,
  height: PropTypes.number,
  weight: PropTypes.number,
  types: PropTypes.array,
};

BaseStats.defaultProps = {
  types: [],
};

export default BaseStats;

export { BaseStats };
