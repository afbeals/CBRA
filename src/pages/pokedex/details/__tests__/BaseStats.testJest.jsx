// External
import React from 'react';
import { render, cleanup } from '@testing-library/react';

// Local
import BaseStats from '../BaseStats';

// tests
describe('BaseStats', () => {
  afterEach(cleanup);
  const mockStats = {};
  beforeEach(() => {
    mockStats.height = 12;
    mockStats.weight = 33;
    mockStats.types = [{ type: { name: 'dragon' } }];
  });
  it('should match BaseStats to snapshot', () => {
    const { getByTestId } = render(<BaseStats pClassname={'parent'} />);
    const baseStats = getByTestId('details-baseStats');
    expect(baseStats).toMatchSnapshot();
  });

  it('should show the correct base stats', () => {
    const { getByText } = render(
      <BaseStats
        pClassname={'parent'}
        height={mockStats.height}
        weight={mockStats.weight}
      />,
    );
    const height = getByText('12');
    const weight = getByText('33');
    expect(weight).toBeTruthy();
    expect(height).toBeTruthy();
  });

  it('should render a dragon type tag', () => {
    const { getByText } = render(
      <BaseStats pClassname={'parent'} types={mockStats.types} />,
    );
    const dragonTag = getByText('dragon');
    expect(dragonTag).toBeTruthy();
    expect(dragonTag.classList.contains('dragon')).toEqual(true);
  });
});
