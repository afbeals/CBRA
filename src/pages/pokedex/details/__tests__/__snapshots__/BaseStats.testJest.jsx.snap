// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseStats should match BaseStats to snapshot 1`] = `
<div
  class="parent__base"
  data-testid="details-baseStats"
>
  <div
    class="parent__base__info"
  >
    <div
      class="section section--info"
    >
      <span
        class="section__title"
      >
        height:
      </span>
      <p
        class="section__content"
      />
    </div>
    <div
      class="section section--info"
    >
      <span
        class="section__title"
      >
        weight:
      </span>
      <p
        class="section__content"
      />
    </div>
    <div
      class="section section--types"
    />
  </div>
</div>
`;
